initialization/initialization.go

```diff
@@ -170,14 +203,38 @@ func (init *Initializer) Initialize(ctx context.Context) error {
 
 	numLabels := uint64(init.opts.NumUnits) * uint64(init.cfg.LabelsPerUnit)
 	fileNumLabels := numLabels / uint64(init.opts.NumFiles)
+	difficulty := shared.PowDifficulty(numLabels)
 
 	init.logger.Info("initialization: starting to write %v file(s); number of units: %v, number of labels per unit: %v, number of bits per label: %v, datadir: %v",
 		init.opts.NumFiles, init.opts.NumUnits, init.cfg.LabelsPerUnit, init.cfg.BitsPerLabel, init.opts.DataDir)
 
 	for i := 0; i < int(init.opts.NumFiles); i++ {
-		if err := init.initFile(ctx, uint(init.opts.ComputeProviderID), i, numLabels, fileNumLabels); err != nil {
+		if err := init.initFile(ctx, i, numLabels, fileNumLabels, difficulty); err != nil {
+			return err
+		}
+	}
+
+	if init.nonce == nil {
+		// continue searching for a nonce
+		res, err := oracle.WorkOracle(
+			oracle.WithComputeProviderID(uint(init.opts.ComputeProviderID)),
+			oracle.WithNodeId(init.nodeId),
+			oracle.WithAtxId(init.atxId),
+			oracle.WithStartAndEndPosition(numLabels, 8*numLabels),
```

Comment:
I updated the code to check until `math.MaxUint64` in batches with the same size that is used to compute leaves. What do you suggest to provide feedback to the user? I don't consider logs as valuable feedback; they are for developers to debug and to include in bug reports imho.

While computing leaves the user feedback is the number of labels written, which is also used by Smapp to display progress.

The problem with how we are now looking for a nonce is that we don't know how long it will take. At least 1 in 50 semshers will have to check > 4x the number of labels as is required to compute leaves with no way to know in advance that it will take this long.

Reply:
Added logs